modules/apps/portal-template/portal-template-react-renderer-impl/src/main/java/com/liferay/portal/template/react/renderer/internal/ReactRendererUtil.java

```diff
@@ -104,6 +140,105 @@ private static Map<String, Object> _prepareProps(
 		return modifiedProps;
 	}
 
+	private static void _renderEcmaScript(
+			ComponentDescriptor componentDescriptor, Map<String, Object> props,
+			AbsolutePortalURLBuilder absolutePortalURLBuilder,
+			DeferredScriptsManager deferredScriptsManager,
+			HttpServletRequest httpServletRequest, String placeholderId,
+			Portal portal, Writer writer)
+		throws IOException {
+
+		StringBundler javascriptSB = new StringBundler(21);
+
+		javascriptSB.append("import {render} from '");
+		javascriptSB.append(
+			absolutePortalURLBuilder.forESModule(
+				"portal-template-react-renderer-impl", "/index.js"
+			).build());
+		javascriptSB.append("';\n");
+
+		String module = componentDescriptor.getModule();
+
+		String[] parts = module.split(" from ");
+
+		javascriptSB.append("import {");
+		javascriptSB.append(_getSymbolName(parts[0]));
+		javascriptSB.append(" as componentModule} from '");
+		javascriptSB.append(
+			absolutePortalURLBuilder.forESModule(
+				parts[1], "/index.js"
+			).build());
+		javascriptSB.append("';\n");
```

Comment:
It is a real shame we don't have a mechanism to make this more readable, something like template literals in java

Reply:
I have tried alternative syntax in the past to make these things more human-friendly, but it was always refactored in the PR process to something inintelligible :shrug: .

We have the TemplateUtil pattern. Maybe I can use that here. I'll try.

Reply:
Sadly, the logic is very complex and I would need to split the templates in several chunks. 

As we will make this simpler in the future, I will "templatize" it later.